### PR TITLE
[WIP] Fixed 26029 -- Configurable file storages backends

### PR DESCRIPTION
**Pull request by sasha0**

### CODE REVIEW

File: django/conf/global_settings.py [hunk 1]
```diff
@@ -587,6 +587,32 @@
     # 'django.contrib.staticfiles.finders.DefaultStorageFinder',
 ]
 
+FILE_STORAGES = {
+    'media': {
```
Comment (sasha0) [inline django/conf/global_settings.py]: @aaugustin @timgraham media backend will have `media` key in settings, not `default`? Looks not quite consistent with the current settings.

Comment (aaugustin) [inline django/conf/global_settings.py]: `DEFAULT_FILE_STORAGE` and `MEDIA_ROOT` aren't consistent anyway. Choose whichever you prefer and explain why :-)

File: django/conf/global_settings.py [hunk 2]
```diff
@@ -587,6 +587,32 @@
...
+        'OPTIONS': {
+            'location': STATIC_ROOT,
+            'base_url': STATIC_URL,
+        },
+        # analogue of STATICFILES_DIRS and STATICFILES_FINDERS
+        'DIRS': [],
+        'FINDERS': [
+            'django.contrib.staticfiles.finders.FileSystemFinder',
+            'django.contrib.staticfiles.finders.AppDirectoriesFinder',
+            # 'django.contrib.staticfiles.finders.DefaultStorageFinder',
```
Comment (aaugustin) [inline django/conf/global_settings.py]: Dirs and finders must be defined in OPTIONS, since they're backend-dependent.

**Comment by aaugustin** [comment]
The approach this PR takes is backwards incompatible. The changes must be implemented with a deprecation path. Look at the introduction of multiple template backends in Django 1.8 for an example. You need pretty much the same strategy.

I would recommend to write down the implementation plan in a DEP and look for consensus before writing too much code. If I recall the last discussion on django-developers correctly, there wasn't an obvious consensus in favor of this change.

**Comment by sasha0** [comment]
@aaugustin thanks for the suggestion - I will do thorough review of approach, taken in multiple template backends functionality and prepare a Django Enhancement Proposal.

**Comment by timgraham** [comment]
Closing this for now.

**Comment by ZuluPro** [comment]
Hello @sasha0,
Do the DEP has been began ?

**Pull request closed without merging**